app/src/main/java/org/p2p/wallet/utils/FragmentExtensions.kt

```diff
@@ -1,11 +1,40 @@
 package org.p2p.wallet.utils
 
+import android.content.ActivityNotFoundException
+import android.content.Intent
+import androidx.core.content.FileProvider
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentFactory
+import org.p2p.uikit.natives.showSnackbarShort
+import org.p2p.wallet.BuildConfig
+import org.p2p.wallet.R
+import timber.log.Timber
+import java.io.File
 import kotlin.reflect.KClass
 
 fun <T : Fragment> Fragment.instantiate(clazz: KClass<T>): Fragment =
     childFragmentManager.fragmentFactory.instantiate(clazz)
 
 fun <T : Fragment> FragmentFactory.instantiate(clazz: KClass<T>): Fragment =
     instantiate(clazz.java.classLoader!!, clazz.java.name)
+
+fun Fragment.openFile(file: File) {
+    val fromFile = FileProvider.getUriForFile(
+        requireContext(),
+        BuildConfig.APPLICATION_ID + ".provider",
+        file
+    )
+
+    val target = Intent(Intent.ACTION_VIEW).apply {
```

Comment:
did you check this method working on Android api 33?

Reply:
I didn't, just took implementation from previous onboarding